packages/dataviews/src/view-list.tsx

```diff
@@ -79,18 +118,42 @@ function ListItem( {
 		}
 	}, [ isSelected ] );
 
+	const { primaryAction, eligibleActions } = useMemo( () => {
+		// If an action is eligible for all items, doesn't need
+		// to provide the `isEligible` function.
+		const _eligibleActions = actions.filter(
+			( action ) => ! action.isEligible || action.isEligible( item )
+		);
+		const _primaryActions = _eligibleActions.filter(
+			( action ) => action.isPrimary && !! action.icon
+		);
+		return {
+			primaryAction: _primaryActions?.[ 0 ],
+			eligibleActions: _eligibleActions,
+		};
+	}, [ actions, item ] );
+
+	const [ isModalOpen, setIsModalOpen ] = useState( false );
+
 	return (
 		<CompositeRow
 			ref={ itemRef }
 			render={ <li /> }
 			role="row"
 			className={ clsx( {
 				'is-selected': isSelected,
+				'is-hovered': isHovered,
 			} ) }
+			onMouseEnter={ handleMouseEnter }
+			onMouseLeave={ handleMouseLeave }
```

Comment:
Super minor, but with these tiny functions I think it's more readable if it's all inlined (especially since they aren't even using `useCallback`:

```js
onMouseEnter={ () => setIsHovered( true ) }
```